Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -182,7 +182,7 @@ private static void OnCarouselPropertyChanged(DependencyObject d, DependencyProp
 
             Carousel carouselControl = (Carousel)d;
 
-            if (e.Property == SelectedIndexProperty)
+            if (e.Property == SelectedIndexProperty && (int)e.NewValue >= 0 && carouselControl.Items.Count > (int)e.NewValue)
```

Comment:
Use the SelectedIndex property instead of (int)e.NewValue. 
```
if ((e.Property == SelectedIndexProperty) && (SelectedIndex >= 0) && (carouselControl.Items.Count > SelectedIndex))
```